src/GraphQL/Validation/Complexity/IFieldComplexityAnalyzer.cs

```diff
@@ -0,0 +1,16 @@
+using GraphQL.Attributes;
+
+namespace GraphQL.Validation.Complexity;
+
+/// <summary>
+/// Represents a field complexity analyzer. Used with the <see cref="ComplexityAttribute"/> to specify
+/// a custom implementation for calculating the complexity of a field. Classes that implement this
+/// interface must have a public parameterless constructor.
```

Comment:
Need analyzer?

Reply:
Ideally that would be good. If developers use the generic version of the attribute then it already has the necessary constraint, but generic attributes are not supported in the .NET Framework. I don’t think it’s a big deal but if you want to add it - either to this pr or as a separate one, I’m all for it.

Reply:
Will add in separate PR

Reply:
Thanks!!!